app/src/org/commcare/tasks/ManageKeyRecordTask.java

```diff
@@ -281,30 +281,29 @@ private void cleanupUserKeyRecords() {
                 newRecord.setType(UserKeyRecord.TYPE_NORMAL);
                 storage.write(newRecord);
             }
+        }
 
+        for (UserKeyRecord recordPendingDelete :

```

Comment:
This was confusing to me: the loop that removes user key records that are set as `TYPE_PENDING_DELETE` was inside another clean-up loop that processes new user key records. This means that it was never getting run. I test this thoroughly in https://github.com/dimagi/commcare-android/pull/1358/files#diff-e0416b2e041f9a9012311119486809ccR115